src/services/point.service.ts

```diff
@@ -1,22 +1,83 @@
+import { Guild, Role, TextChannel, MessageEmbed, EmojiIdentifierResolvable } from 'discord.js';
 import { IUserPoints, Maybe } from '../common/types';
 import { PointsDocument, PointsModel } from '../schemas/points.schema';
 import { GuildService } from './guild.service';
+import Constants from '../common/constants';
+import { UserService } from './user.service';
+import ms from 'ms';
 
 export class PointService {
   private _guild: Guild;
+  private _userservice: UserService;
+  private _tacoKingRole: Role;
+  private _tacoKingEmoji: Maybe<EmojiIdentifierResolvable> = null;
 
-  constructor(private _guildService: GuildService) {
+  constructor(private _guildService: GuildService, private _userService: UserService) {
     this._guild = this._guildService.get();
+    this._userservice = this._userService;
+    this._tacoKingRole = this._guildService.getRole(Constants.Roles.TacoKing);
+    this._tacoKingEmoji = this._guildService.getEmoji('tacoking') ?? '🌮';
   }
 
   public async awardPoints(id: Maybe<string>, amount: number) {
     if (!id) {
       return;
     }
 
+    // cache the #1 point holder before points are awarded
+    const [{ userID: kingBeforeAwardID }] = await this.getTopPoints(1);
+
     const userDoc = await this.getUserPointDoc(id);
     await PointsModel.updateOne(userDoc, { $inc: { numPoints: amount } });
+
+    const [{ userID: kingAfterAwardID }] = await this.getTopPoints(1);
+
+    // return if the king hasnt changed
+    if (kingBeforeAwardID === kingAfterAwardID) {
+      return;
+    }
+
+    // get the old and new king members
+    const [prevKingUser, newKingUser] = [kingBeforeAwardID, kingAfterAwardID].map((c) =>
+      this._userservice.getMember(c)
+    );
+
+    // remove the taco king role from the previous king
+    // and give the role to the new king
+    await Promise.all([
+      prevKingUser?.roles.remove(this._tacoKingRole),
+      newKingUser?.roles.add(this._tacoKingRole),
+    ]);
+
+    // update the crowning date for the new king
+    await PointsModel.updateOne(
+      { userID: kingAfterAwardID, guildID: this._guild.id },
+      { lastKingCrowning: new Date() }
+    );
+
+    const crownedTime = await this.getCrownedTime(kingBeforeAwardID);
+
+    // we are gonna send a message in the games channel
+    const gamesChan = this._guildService.getChannel(Constants.Channels.Public.Games);
+
+    const embed = new MessageEmbed()
+      .setTitle(`${this._tacoKingEmoji} Taco King Overthrown! ${this._tacoKingEmoji}`)
+      .setDescription(
+        `${prevKingUser?.user ?? 'The old king'} is no longer the Taco King!\n${
+          newKingUser?.user ?? 'Someone else'
+        } is the new ${this._tacoKingRole}!`
+      )
+      .setColor(this._tacoKingRole.color);
+
+    if (crownedTime > ms('1s')) {
+      embed.setFooter({
+        text: `They were the king for ${ms(crownedTime, {
+          long: true,
+        })}!`,
+      });
+    }
+
+    await (gamesChan as TextChannel).send({ embeds: [embed] });
```

Comment:
doesnt need to be in this diff but

at some point its prolly worth to make a `getTextChannel` method so we dont have to always force cast

Reply:
Overkill, but you could also use generics in the function definition to provide what type of channel you want